### PR TITLE
tse new feature correcction

### PR DESCRIPTION
**Pull request by eng-eng11**
## Project 1: Bubble Sort
Sorting algorithms are some of the earliest that you typically get exposed to in Computer Science. It may not be immediately obvious how important they are, but it shouldn’t be hard to think of some examples where your computer needs to sort some massive datasets during everyday operations.

### CODE REVIEW

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Hi Falguniislam, :wave: 

After going through your code, I made the following observation:

- [x] Your first function in your new bubblesort.rb file, does not sort the array. I also observed you did not loop with ' _i ' index you inserted into "pipe". Kindly fix that. :slightly_smiling_face: 

- [x] Also, work on the linter errors.

However, the bubblesort.rb file you deleted, the function sorted the array, properly.

``` 
def bubble_sort(arr)
  arr.each do
    i = 0
    swap = 0
    while i < arr.length - 1
      if arr[i] > arr[i + 1]
        temp = arr[i]
        arr[i] = arr[i + 1]
        arr[i + 1] = temp
      end
      i += 1
      swap += 1 
    end
    break if swap.zero?
  end
end

```

After implementing this change submit another form for a code review.

#### ``` Have a nice day ``` :slightly_smiling_face: 

@ Uchechukwu Henry Inyama

File: README.md [hunk 1]
```diff
@@ -1,10 +1,64 @@
...
+        temp = arr[i]
+        arr[i] = arr[i + 1]
+        arr[i + 1] = temp
+      end
+      i += 1
+      swap += 1
+    end
+    break if swap.zero?
+  end
+  puts arr.join(' ')
```
Comment (maelfosso) [inline README.md:36]: You must return the array not print it

File: README.md [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
+    (0...a).each do |i|
+      test = yield(array[i], array[i + 1])
+      next unless test.positive?
+
+      temp = array[i]
+      array[i] = array[i + 1]
+      array[i + 1] = temp
+    end
+  end
+  puts array.join(' ')
```
Comment (maelfosso) [inline README.md:54]: You must return the array not print it

File: README.md [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
+end
+
+bubble_sort([4,3,78,2,0,2])
+# => [0,2,2,3,4,78]
+
+bubble_sort_by(["hi","hello","hey"]) do |left,right|
+  left.length - right.length
+end
```
Comment (maelfosso) [inline README.md:62]: Not needed

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @falguniislam @davidauza-engineer 

Nice improvements :smile: 
To make the project approved, you need to just change some lines. Please, read the comments below and **correct the sticklers errors**

Also, **Follow the Git Flow** otherwise the project won't be approved.
- Do not work directly in the master branch. Create the development branch and a specific feature-branch (give it a descriptive name) for every feature you implement. When your feature is ready and approved you can merge feature-branch to the development branch and when the release is ready create a pull request from development to master.
- Read more [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/)

Kindly make the changes and submit a new code review request.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com)

File: README.md (hunk 1 above)
Comment (davidauza-engineer) [reply README.md:36]: Done :)

File: README.md (hunk 2 above)
Comment (davidauza-engineer) [reply README.md:54]: Done :)

File: README.md (hunk 3 above)
Comment (davidauza-engineer) [reply README.md:62]: Successfully removed :)

**Review by davidauza-engineer — COMMENTED** [review]
All suggested changes were done. Thank you so much. Please let us know if everything is OK now :)

**Comment by davidauza-engineer** [comment]
Accidentally closed, reopening...

**Review by maelfosso — APPROVED** [review]
# You did it :smile:  Congratulations :clap: 
# The project is approved :+1: 

You can merge the project now.
Good luck with the next project

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com)

**Comment by davidauza-engineer** [comment]
Thanks a lot @maelfosso 👍

**Pull request merged**